Assets/StarterAssets/FirstPersonController/Scripts/FirstPersonController.cs

```diff
@@ -162,33 +163,34 @@ private void Move()
 
             if (sprinting)
             {
-                stamina -= 1 * Time.deltaTime;
+                stamina -= _staminaRechargeMultiplier * Time.deltaTime;
             } else if(stamina < maxStamina && !isFatigued)
             {
-                stamina += 1 * Time.deltaTime;
+                stamina += _staminaRechargeMultiplier * Time.deltaTime;
             }
 
             if (isFatigued)
             {
-                if(FatigueTimer+1f < Time.time)
+                if(FatigueTimer < Time.time)
                 {
-                    fatigueRecharge += 1 * Time.deltaTime;
+                    fatigueRecharge += _staminaRechargeMultiplier * Time.deltaTime;
                 }
                 if(fatigueRecharge >= fatigueTimer)
                 {
-                    //Debug.Log("Recharge complete");
+                    // Recharge complete
                     isFatigued = false;
-                    stamina = 0;
                     fatigueRecharge = 0;
                 }
+				stamina += _staminaRechargeMultiplier * Time.deltaTime;
             }
 
 
 			if (stamina < 0 && !isFatigued)
 			{
+                // Is fatigued
+				// TODO: Add a threshold value so that the player becomes fatigued when ending a sprint below a certain value as well?
```

Comment:
The approach in this PR uses the "fatigueTimer" as a threshold, so this "TODO" is not valid anymore. Remove it if the code is to be merged and used.